pymchelper/readers/shieldhit.py

```diff
@@ -9,9 +9,372 @@
+    ext_nproj = 0xCC0F
+    ext_ptvdose = 0xCC10
+    ixfirs = 0xCC11
+
+    # Group 0xCE00 - 0xCEFF : CT specific tags
+    ct_ang = 0xCE00   # holds two doubles with the couch and gantry angle
+    ct_icnt = 0xCE01  # holds three
+    ct_len = 0xCE02   # holds three
+
+    # Group 0xEE00 - 0xEEFF : Estimator specific tags
+    # estimator specific tags
+    est_geotyp = 0xEE00    # may differ from det_geotyp in case of spc
+    est_pages = 0xEE01     # number of detectors / pages for this estimator
+
+    # Group 0xDD00 - 0xDDFF : Detector/page specific tags
+    det_geotyp = 0xDD00      # may differ from est_geotyp in case of spc
+    det_nbin = 0xDD01        # idet(1-3) (len=3) number of bins x,y,z
+    det_part = 0xDD02        # idet(4) particle type which was scored
+    det_dtype = 0xDD03       # idet(5) detector type
+    det_partz = 0xDD04       # idet(6)
+    det_parta = 0xDD05       # idet(7)
+    det_dmat = 0xDD06        # idet(8)
+    det_nbine = 0xDD07       # idet(9)
+    det_difftype = 0xDD08    # idet(10)
+    det_zonestart = 0xDD09   # idet(11)
+    det_dsize = 0xDD0A       # idet(12)
+    det_dsizexyz = 0xDD0B    # idet(13)
+
+    det_xyz_start = 0xDD0C   # det(1-3)
+    det_xyz_stop = 0xDD0D    # det(4-6)
+    det_dif_start = 0xDD0E   # det(7)
+    det_dif_stop = 0xDD0F    # det(8)
+    det_voxvol = 0xDD10      # det(9)
+
+    det_data = 0xDDBB        # data block
+
+    # Group 0xAA00 - 0xAAFF : Runtime variables
+    rt_nstat = 0xAA00        # number of actually simulated particles
+    rt_time = 0xAA01         # [usignend long int] optional runtime in seconds
+
+
+# for future use
+# mapping = {SHBDOTagID.shversion: "mc_code_version",
+#           SHBDOTagID.filedate: "filedate",
+#           SHBDOTagID.user: "user",
+#           SHBDOTagID.host: "host",
+#           SHBDOTagID.rt_nstat: "nstat",
+#           SHBDOTagID.det_dtype: "dettyp",
+#           SHBDOTagID.est_geotyp: "geotyp",
+#           SHBDOTagID.det_xyz_start: ("xmin", "ymin", "zmin"),
+#           SHBDOTagID.det_xyz_stop: ("xmax", "ymax", "zmax"),
+#           SHBDOTagID.det_nbin: ("nx", "ny", "nz")}
+
+
 class SHBinaryReader:
     """
-    Reads binary output files generated by SHIELDHIT12A code.
+    Reads binary output files generated by SHIELD-HIT12A code.
+    """
+    def __init__(self, filename):
+        self.filename = filename
+
+    def test_version_0p6(self):
+        sh_bdo_magic_number = b'xSH12A'
+        with open(self.filename, "rb") as f:
+            d1 = np.dtype([('magic', 'S6'),
+                           ('end', 'S2'),
+                           ('vstr', 'S16')])
+            x = np.fromfile(f, dtype=d1, count=1)
+
+            # if magic string is present, deep check for version number
+            if sh_bdo_magic_number == x['magic'][0]:
+                vmaj, vmin = x['vstr'][0].decode('ASCII').split('.')
+                return (sh_bdo_magic_number == x['magic'][0]) and (int(vmaj) >= 0) and (int(vmin) >= 6)
```

Comment:
I haven't slept much but is version comparison OK ?
what if we arrive at SH version 1.0 ?
Then vmaj = 1, vmin = 0 and this line will return false.